src/app/public/testing/sort/sort-fixture.ts

```diff
@@ -0,0 +1,178 @@
+ * of a component, such as changing its DOM structure.
+ */
+export class SkySortFixture {
+
+  /**
+   * The active sort menu item, if one exists. Menu items are only available when the menu dropdown
+   * is open. If the menu dropdown is closed, this property will be undefined.
+   */
+  public get activeMenuItem(): SkySortFixtureMenuItem {
+    return this.menuItems?.find(x => x.isActive);
+  }
+
+  /**
+   * The sort menu's properties.
+   */
+  public get menu(): SkySortFixtureMenu {
+    return {
+      buttonText: SkyAppTestUtility.getText(this.getSortButtonTextEl()),
+      isOpen: this.getDropdownMenuEl() !== null
+    };
+  }
+
+  /**
+   * The properties of each sort menu item. Menu items are only available when the menu dropdown
+   * is open. If the menu dropdown is closed, this property will be undefined.
+   */
+  public get menuItems(): SkySortFixtureMenuItem[] {
+    // Return undefined when we can't determine what the options are.
+    // We do this to avoid any confusion with an empty set of options.
+    if (!this.menu.isOpen) {
+      return;
+    }
+
+    return this.getSortItems()
+      .map((item: HTMLElement) => {
+        const itemButton = item.querySelector('button');
+
+        return {
+          isActive: item.classList.contains('sky-sort-item-selected'),
+          text: SkyAppTestUtility.getText(itemButton)
+        };
+      });
+  }
+
+  private _debugEl: DebugElement;
+
+  constructor(
+    private fixture: ComponentFixture<any>,
+    skyTestId: string
+  ) {
+    this._debugEl = SkyAppTestUtility.getDebugElementByTestId(fixture, skyTestId, 'sky-sort');
+  }
+
+  /**
+   * Closes the sort dropdown menu if it isn't closed already.
+   */
+  public async closeMenu(): Promise<void> {
+
+    // if the menu is already closed, do nothing
+    if (!this.menu.isOpen) {
+      return;
+    }
+
+    const menu = this.getDropdownButtonEl();
+    if (menu !== undefined && !menu.disabled) {
+      menu.click();
+
+      this.fixture.detectChanges();
+      await this.fixture.whenStable();
+
+      this.fixture.detectChanges();
+      return this.fixture.whenStable();
+    }
+  }
+
+  /**
+   * Opens the sort dropdown menu if it isn't open already.
+   */
+  public async openMenu(): Promise<void> {
+    // if the menu is already open, do nothing
+    if (this.menu.isOpen) {
+      return;
+    }
+
+    const menu = this.getDropdownButtonEl();
+
+    if (menu !== undefined && !menu.disabled) {
+      menu.click();
+
+      this.fixture.detectChanges();
+      await this.fixture.whenStable();
+
+      this.fixture.detectChanges();
+      return this.fixture.whenStable();
+    }
+  }
+
+  /**
+   * Ensures the sort menu is open and selects the menu item with the specified text,
+   * if a matching item is available.
+   * @param menuItemText The text of the menu item to select.
+   */
+  public async selectMenuItemByText(menuItemText: string): Promise<void> {
```

Comment:
Your previous comment was a two-parter. I addressed the naming issue here already.

As to why I chose text instead of an index, it was mainly for convenience. I can certainly add a selectByIndex in addition, or remove selectByText if it's not something we want to support.

Reply:
Annnnnd now we support both.